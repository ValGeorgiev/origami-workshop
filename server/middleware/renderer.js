import React from 'react'
import ReactDOMServer from 'react-dom/server'
import { StaticRouter } from 'react-router-dom'

// import our main App component
import App from '../../src/App';
import Navigation from '../../src/navigation';

import path from "path"
import fs from "fs"

export default (req, res, next) => {

    // point to the html file created by CRA's build tool
    const filePath = path.resolve(__dirname, '..', '..', 'build', 'index.html');

    fs.readFile(filePath, 'utf8', (err, htmlData) => {
        if (err) {
            console.error('err', err);
            return res.status(404).end()
        }

        // render the app as a string
        const html = ReactDOMServer.renderToString(
          <App origamis={req.origamis} user={req.user}>
            <StaticRouter>
              <Navigation />
            </StaticRouter>
          </App>
        );

        const htmlWithState = htmlData.replace(
          'window.__STATE__={}',
          `window.__STATE__=${JSON.stringify({
            user: req.user,
            origamis: req.origamis
          })}`
        )

        // inject the rendered app into our html and send it
        return res.send(
          htmlWithState.replace(
                '<div id="root"></div>',
                `<div id="root">${html}</div>`
            )
        );
    });
}