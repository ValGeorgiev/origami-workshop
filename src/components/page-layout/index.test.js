import React from 'react'
import renderer from 'react-test-renderer'
import PageLayout from '.'
import TestingEnvironment from '../../test-utils/router'
jest.mock('../header', () => 'Header');
jest.mock('../aside', () => 'Aside');
jest.mock('../footer', () => 'Footer');

describe('PageLayout Component', () => {
  it('should render pagelayout component', () => {
  
    
    const tree = renderer.create(
      <TestingEnvironment value={{
        user: {
          loggedIn: true,
          id: '123'
        }
      }}>
        <PageLayout />
      </TestingEnvironment>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})