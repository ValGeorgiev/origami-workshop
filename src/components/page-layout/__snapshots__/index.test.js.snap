// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout Component should render pagelayout component 1`] = `
<div
  className="app"
>
  <Header />
  <div
    className="container"
  >
    <Aside />
    <div
      className="inner-container"
    />
  </div>
  <Footer />
</div>
`;
